## **Case-Study:**
<p>
<strong>CrossBorder</strong> Films is one of the leading DVD rental companies that offers film rental services
across different countries. Having completed the 10Alytics SQL training module, the head of
data and Analytics of CrossBorder Films has hired you as an analyst to help them make sense of
their business data. They have sent you their dvdrental database(Same dvdrental database you
have been working with) to work with.
</p>
<p>
They require you to answer the following questions about their film rental business
performance:
</p>

**Question 1**
<p>Who are the top loyal customers of CrossBorder Films and where do they live?</p>

In [3]:
%load_ext sql

In [32]:
%sql postgresql://postgres:Smiley12@localhost/dvdrental

In [81]:
%%sql select c.customer_id, co.country, count(r.rental_id) as no_of_rentals, c.create_date
from rental r join customer c
on r.customer_id = c.customer_id 
join address a on c.address_id = a.address_id
join city ct on a.city_id = ct.city_id
join country co on ct.country_id = co.country_id
where c.active = 1
group by 1, 2
order by 3 desc, 4 desc
limit 10

 * postgresql://postgres:***@localhost/dvdrental
10 rows affected.


[(148, 'Runion', 46, datetime.date(2006, 2, 14)),
 (526, 'United States', 45, datetime.date(2006, 2, 14)),
 (236, 'Philippines', 42, datetime.date(2006, 2, 14)),
 (144, 'Belarus', 42, datetime.date(2006, 2, 14)),
 (75, 'Taiwan', 41, datetime.date(2006, 2, 14)),
 (469, 'Spain', 40, datetime.date(2006, 2, 14)),
 (197, 'China', 40, datetime.date(2006, 2, 14)),
 (178, 'Brazil', 39, datetime.date(2006, 2, 14)),
 (468, 'India', 39, datetime.date(2006, 2, 14)),
 (137, 'Netherlands', 39, datetime.date(2006, 2, 14))]

**Question 2** 
<p>
Generate a report of film rental performance over the years looking at how many
films were returned late, early and on time?
</p>

In [79]:
%%sql with rental_summary as (select *, date_part('day', return_date - rental_date) as date_diff
from rental),
date_summary as (select rental_duration, date_diff,
    case
        when rental_duration > date_diff then 'early'
        when rental_duration < date_diff then 'late'
        else 'on time'
    end return_status
from film f
join inventory i
using(film_id)
join rental_summary
using(inventory_id))
select return_status, count(*) as total_no_of_films
from date_summary
group by 1
order by 2 desc;

 * postgresql://postgres:***@localhost/dvdrental
3 rows affected.


[('early', 7738), ('late', 6403), ('on time', 1903)]

**Question 3**
<p>What are the top 3 highest in-demand genres of film over the years and what are
their total sales?
</p>

In [80]:
%%sql select ca.name as genre, count(r.customer_id) as no_of_rentals, sum(pa.amount) as total_amount
from payment pa
join rental r
using(rental_id)
join inventory i
using(inventory_id)
join film f
using(film_id)
join film_category fc
using(film_id)
join category ca
using(category_id)
group by 1
order by no_of_rentals desc
limit 3

 * postgresql://postgres:***@localhost/dvdrental
3 rows affected.


[('Sports', 1081, Decimal('4892.19')),
 ('Animation', 1065, Decimal('4245.31')),
 ('Action', 1013, Decimal('3951.84'))]

**Question 4**
<p>Identify the top 5 countries where CrossBorder Films had the highest volume of
sales along-side the countries where they had the largest customer base.
</p>

In [67]:
%%sql select co.country, count(distinct cu.customer_id) as customer_base, sum(pa.amount) as total_sales
from country co 
join city ci
using(country_id)
join address ad
using(city_id)
join customer cu
using(address_id)
join payment pa
using(customer_id)
group by 1
order by 2 desc
limit 5

 * postgresql://postgres:***@localhost/dvdrental
5 rows affected.


[('India', 60, Decimal('6034.78')),
 ('China', 53, Decimal('5251.03')),
 ('United States', 36, Decimal('3685.31')),
 ('Japan', 31, Decimal('3122.51')),
 ('Mexico', 30, Decimal('2984.82'))]

**Question 5**
<p>Identify the top 5 countries where CrossBorder Films had the lowest customer
base.
</p>

In [69]:
%%sql select co.country, count(distinct cu.customer_id) as customer_base
from country co 
join city ci
using(country_id)
join address ad
using(city_id)
join customer cu
using(address_id)
group by 1
order by 2 
limit 5

 * postgresql://postgres:***@localhost/dvdrental
5 rows affected.


[('Armenia', 1),
 ('American Samoa', 1),
 ('Afghanistan', 1),
 ('Anguilla', 1),
 ('Bahrain', 1)]

## **Question 6**
<p>For each of the questions above, explain in one paragraph what insight can be concluded from
the result of your query.
</p>

### Insight 1

The customer with ID 148 is considered the most loyal, based on their number of rentals and the date of their registration. This customer has been a part of the business since its inception and has the highest number of rentals to their name.

### Insight 2

A total of 16,044 movies were returned. 48.23% were returned early, 39.90% were returned on time while 11.86% were returned late.

### Insight 3

CrossBorder offers 16 distinct genres for its customers to choose from. The sports category is the most popular in terms of rentals, and it also generates the highest total sales revenue.

### Insight 4

The largest customer base for the company is in India, with 60 customers, and the highest total sales revenue is also recorded in India

### Insight 5

Armania has the lowest customer base 